### PR TITLE
Android guidelines - Support for captions for prerecorded and live-stream media

### PR DESCRIPTION
**Pull request by terezijaumiljanovic**
Add guidelines for captions in prerecorded and live-stream content [#29](https://github.com/infinum/accessibility-mobile-standards/issues/29)

### CODE REVIEW

**Review by zavrskina — APPROVED** [review]
Good job! 👏 🎉

File: docs/guidelines/platforms/android/guideline_percievable_android.md [hunk 1]
```diff
@@ -23,23 +23,60 @@ If an element exists on the UI only for decorative purposes it is recommended th
...
 
-- If a video presents information that is vital to completing workflow, provide the same content in an alternative format, such as a transcript
+To view captions in media that supports them, users must enable the _Show captions_ option. This accessibility feature can be found under **Settings** > **Accessibility** > **Caption preferences**. On this screen, users can also adjust caption size and style; however, these preferences may not work with media apps that don’t support Caption Preferences.
 
-:no_entry_sign: **Failure criteria**
+##### Custom solution
+
+If a different player is used instead of `ExoPlayer`, the status of closed captions, along with the font scale and caption style, can be checked programmatically using the `CaptioningManager`. When the _Show captions_ option is enabled, captions should be turned on in the selected player.
```
Comment (Aksi0m) [inline docs/guidelines/platforms/android/guideline_percievable_android.md:46]: A QoL improvement would be to add the link to the class docs. https://developer.android.com/reference/android/view/accessibility/CaptioningManager

File: docs/guidelines/platforms/android/guideline_percievable_android.md [hunk 2]
```diff
@@ -23,23 +23,60 @@ If an element exists on the UI only for decorative purposes it is recommended th
...
+
+### Captions support for live media (WCAG 1.2.4 - Level AA)
+
+This guideline covers [1.2.4 Captions (Live) - Level AA](https://www.w3.org/WAI/WCAG22/quickref/#captions-live) of the WCAG standard.
+
+In some cases, live media is used in the application. To ensure accessibility for all users, providing captions for live content is important.
+
+:white_check_mark: **Success criteria**
+
+For all live media in the application, captions should be available through the user interface. This can be achieved in two ways:
```
Comment (Aksi0m) [inline docs/guidelines/platforms/android/guideline_percievable_android.md:74]: I would maybe add the exemption from the docs here, just to be a bit more cautious about what the application should/shouldn't handle. 
Something like this:

All live media in the application should offer captions through the user interface, available as either open or closed captions. For two-way multimedia calls between individuals, accessibility requirements are the responsibility of content providers, not the application.

**Open captions**: ...

**Review by Aksi0m — APPROVED** [review]
Looks good! Thanks for the efforts 🎖️

File: docs/guidelines/platforms/android/guideline_percievable_android.md [hunk 3]
```diff
@@ -23,23 +23,60 @@ If an element exists on the UI only for decorative purposes it is recommended th
...
+class CaptionUtils(context: Context) {
+    
+   private val captioningManager: CaptioningManager =
+       context.getSystemService(Context.CAPTIONING_SERVICE) as CaptioningManager
+    
+   fun areCaptionsEnabled(): Boolean = captioningManager.isEnabled
+    
+   fun getCaptionFontScale(): Float = captioningManager.fontScale
+    
+   fun getCustomCaptionStyle(): CaptionStyle = captioningManager.userStyle
```
[the remark this answers is not in the document]
Comment (terezijaumiljanovic) [reply docs/guidelines/platforms/android/guideline_percievable_android.md:62]: It is a class in the `CaptioningManager` that defines the appearance of captions displayed on the screen such as foregroundColor, backgroundColor, edgeType, edgeColor and windowColor.

**Pull request merged**